### PR TITLE
fix joyent disk mounting and other confirm issues

### PR DESCRIPTION
**Pull request by duddun**

### CODE REVIEW

File: lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb [hunk 1]
```diff
@@ -158,23 +164,27 @@ def confirm(inputmap)
...
           begin
-            mounted = false
-            ssh_exec!(ssh, 'if grep "vdb /data " /proc/mounts ; then /bin/false ; fi', 'Checking if /dev/vdb mounted already')
+            data_mounted = false
+            ssh_exec!(ssh, 'if grep "vdb /data " /proc/mounts ; then echo "/dev/vdb is mounted"; else /bin/false ; fi', 'Checking in /proc/mounts whether /dev/vdb mounted already')
+            data_mounted = true
           rescue
-            mounted = true
+            log.debug 'Disk /dev/vdb is not mounted to /data'
+            log.debug "data_mounted = #{data_mounted}"
```
Comment (dereklwood) [inline lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb]: remove this log line... not needed

Comment (nitinmotgi) [inline lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb]: +1

###
Random auto-corrects and typos are my special gift to you. When I forward they are from others.

File: lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb [hunk 2]
```diff
@@ -158,23 +164,27 @@ def confirm(inputmap)
...
+          unless data_mounted
+            # disk isn't mounted at /data, could be mounted elsewhere (e.g. /mnt)
             begin
-              # Are we mounted?
-              ssh_exec!(ssh, 'mount | grep ^/dev/vdb 2>&1 >/dev/null', 'Checking if /dev/vdb is unmounted')
+              # Are we mounted? # if we are, set unmount to true
+              ssh_exec!(ssh, 'if mount | grep "^/dev/vdb " ; then echo "Disk /dev/vdb is mounted" ; else /bin/false ; fi', 'Confirm /dev/vdb is mounted')
               unmount = true
             rescue
+              mounted = false
```
Comment (dereklwood) [inline lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb]: this doesnt appear to be used.  pls confirm and delete.

File: lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb [hunk 3]
```diff
@@ -130,6 +132,11 @@ def confirm(inputmap)
       Net::SSH.start(bootstrap_ip, @task['config']['ssh-auth']['user'], @credentials) do |ssh|
         ssh_exec!(ssh, 'ping -c1 www.opscode.com', 'Validating external connectivity and DNS resolution via ping')
         ssh_exec!(ssh, "#{sudo} hostname #{@task['config']['hostname']}", 'Temporarily setting hostname')
+        # let us check and make sure firstboot is done running
```
Comment (dereklwood) [inline lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb]: `# Check and` ...

File: lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb [hunk 4]
```diff
@@ -158,23 +164,27 @@ def confirm(inputmap)
...
             begin
-              # Are we mounted?
-              ssh_exec!(ssh, 'mount | grep ^/dev/vdb 2>&1 >/dev/null', 'Checking if /dev/vdb is unmounted')
+              # Are we mounted? # if we are, set unmount to true
+              ssh_exec!(ssh, 'if mount | grep "^/dev/vdb " ; then echo "Disk /dev/vdb is mounted" ; else /bin/false ; fi', 'Confirm /dev/vdb is mounted')
               unmount = true
             rescue
+              mounted = false
               log.debug 'Disk /dev/vdb is not mounted'
+              unmount = false
```
Comment (dereklwood) [inline lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb]: move this to just before the ssh_exec!.  for consistency with `data_mounted` above, and its good to minimize logic in a rescue block

**Comment by dereklwood** [comment]
some minor comments

File: lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb (hunk 1 above)
Comment (duddun) [inline lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/joyent.rb]: Nice catch!  I obviously had that for troubleshooting the issue.

**Comment by dereklwood** [comment]
:+1:

**Comment by duddun** [comment]
Addresses the following issues:
https://issues.cask.co/browse/COOPR-729
https://issues.cask.co/browse/COOPR-730

**Pull request merged**